src/pybind/mgr/orchestrator/module.py

```diff
@@ -964,6 +964,24 @@ def _set_prometheus_access_info(self, username: Optional[str] = None, password:
         except ArgumentError as e:
             return HandleCommandResult(-errno.EINVAL, "", (str(e)))
 
+    @_cli_write_command('orch prometheus set-target')
+    def _set_prometheus_targets(self, url: Optional[str] = None, inbuf: Optional[str] = None) -> HandleCommandResult:
+        try:
+            completion = self.set_prometheus_targets(url)
+            result = raise_if_exception(completion)
+            return HandleCommandResult(stdout=json.dumps(result))
+        except ArgumentError as e:
```

Comment:
I don't think this or the remove version call anything that can raise an `ArgumentError`? Afaict, that is only raised by `_get_credentials` in this same file, which isn't being called here.

Reply:
Updated the PR